pkg/reconciler/image/reconcile_build.go

```diff
@@ -71,27 +76,49 @@ func (c *Reconciler) reconcileBuild(ctx context.Context, image *buildapi.Image,
 }
 
 func noScheduledBuild(buildNeeded corev1.ConditionStatus, builder buildapi.BuilderResource, build *buildapi.Build, sourceResolver *buildapi.SourceResolver) corev1alpha1.Conditions {
```

Comment:
This function reads a little bit awkward with all the ifs that immediately return. What do you think of converting it to a switch statement to make it more explicitly layout the possible causes:

```go
func noScheduledBuild(buildNeeded corev1.ConditionStatus, builder buildapi.BuilderResource, build *buildapi.Build, sourceResolver *buildapi.SourceResolver) corev1alpha1.Conditions {
	var (
		status  corev1.ConditionStatus
		reason  string
		message string
	)

	switch {
	case !builder.Ready():
		status = corev1.ConditionFalse
		reason = buildapi.BuilderNotReady
		message = builderError(builder)
	case buildNeeded == corev1.ConditionUnknown:
		status = corev1.ConditionUnknown
		reason = ResolverNotReadyReason
		if sourceResolver.Ready() {
			message = "Build status unknown"
		} else {
			message = fmt.Sprintf("SourceResolver %s is not ready", sourceResolver.GetName())
		}
	case !build.Status.GetCondition(corev1alpha1.ConditionSucceeded).IsTrue():
		status = unknownStatusIfNil(build.Status.GetCondition(corev1alpha1.ConditionSucceeded))
		reason = BuildFailedReason
		message = defaultMessageIfNil(build.Status.GetCondition(corev1alpha1.ConditionSucceeded), "Last build did not succeed")
	default:
		status = unknownStatusIfNil(build.Status.GetCondition(corev1alpha1.ConditionSucceeded))
		reason = UpToDateReason
		message = defaultMessageIfNil(build.Status.GetCondition(corev1alpha1.ConditionSucceeded), "Last build succeeded")
	}

	return corev1alpha1.Conditions{
		{
			Type:               corev1alpha1.ConditionReady,
			Status:             status,
			Reason:             reason,
			Message:            message,
			LastTransitionTime: corev1alpha1.VolatileTime{Inner: metav1.Now()},
		},
		builderCondition(builder),
	}
}

```